internal/wrphandlers/qos/priority_queue.go

```diff
@@ -69,16 +78,42 @@ func (pq *priorityQueue) drop() {
 func (pq *priorityQueue) Len() int { return len(pq.queue) }
 
 func (pq *priorityQueue) Less(i, j int) bool {
-	return pq.queue[i].QualityOfService > pq.queue[j].QualityOfService
+	iQOS := pq.queue[i].msg.QualityOfService
+	jQOS := pq.queue[j].msg.QualityOfService
+	iTimestamp := pq.queue[i].timestamp
+	jTimestamp := pq.queue[j].timestamp
+
+	// Determine what will be used as a priority tie breaker.
+	var tieBreaker bool
+	switch pt := pq.priority; pt {
```

Comment:
Instead of a switch here, make a function that is set by the options & validated when the PQ is created.  That saves a check here each time & the need to handle a panic scenario.

Reply:
Oh ok, it didn't click the first time but now I get it 🙂